lib/task.js

```diff
@@ -97,15 +98,20 @@ function factory(
         self.successStates = [TaskStates.Succeeded];
         // hint to whatever is running the task when it has failed
         self.failedStates = [TaskStates.Failed, TaskStates.Timeout, TaskStates.Cancelled];
-
+   
+        var server;
+        if (_.has(context, 'proxy')) {
+            server = context.proxy;
+        } else {
+            server = 'http://%s:%s'.format(
+                Task.configCache.apiServerAddress,
+                Task.configCache.apiServerPort
```

Comment:
A typical endpoint look like below (in monorail.json or config.json). I would suggest to get the address and port from the endpoint with northbound-api-router. 

```
"httpEndpoints": [
        {
            "address": "0.0.0.0",
            "port": 8080,
            "httpsEnabled": false,
            "proxiesEnabled": true,
            "authEnabled": false,
            "routers": "northbound-api-router"
        },
        {
            "address": "172.31.128.1",
            "port": 9080,
            "httpsEnabled": false,
            "proxiesEnabled": true,
            "authEnabled": false,
            "routers": "southbound-api-router"
        }
    ]
```

Since apiServerAddress and apiServerPort will also work, I agree this is not a gating item.